sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/CallMethodViaReflection.scala

```diff
@@ -55,7 +55,7 @@ import org.apache.spark.util.Utils
 case class CallMethodViaReflection(children: Seq[Expression])
   extends Expression with CodegenFallback {
 
-  override def prettyName: String = "reflect"
+  override def prettyName: String = getTagValue(FunctionRegistry.FUNC_ALIAS).getOrElse("reflect")
```

Comment:
Does this change from `reflect` to `java_method` for `CallMethodViaReflection`? Why not keep `reflect`?

Reply:
@viirya, it changes from `reflect` to `java_method` when `java_method` is used, which I believe is correct. It will still show `reflect` when `reflect` is used.

```scala
scala> spark.range(1).selectExpr("java_method('java.util.UUID', 'randomUUID')").show()
+---------------------------------------+
|java_method(java.util.UUID, randomUUID)|
+---------------------------------------+
|                   c1b2961e-ccc4-4ad...|
+---------------------------------------+


scala> spark.range(1).selectExpr("reflect('java.util.UUID', 'randomUUID')").show()
+-----------------------------------+
|reflect(java.util.UUID, randomUUID)|
+-----------------------------------+
|               587b9a03-63f1-424...|
+-----------------------------------+
```

Reply:
Yea, if a code relies on the output column name of `selectExpr("java_method(...)")`, it could be broken as it is changed to `java_method(...)` now. This is correct, I think. If we don't guarantee on the output column name, it is ok.

Reply:
I don't think we guarantee the auto-generated alias name. It's very fragile and can get changed super easily without noticing. AFAIK it has been changed many times in the past.